### DISCUSSION TITLE
Check if Scoop Exists

### DISCUSSION THREAD

**Discussion by bahyway** [ScoopInstaller/Scoop#5421] [General]
Hi,
Do you know how can I check if Scoop already exists in the os or not, the reason for my question is that I tried to run a scoop script for the second time and I got the following Error:

Scoop is already installed. Run 'scoop update' to get the latest version.
Abort.

I need to run the Scoop script as much as I can when I add new apps to the script, the silently run of the script needs to be smooth and not buggy. 

any idea?

**Comment by rashil2000** [comment]
You can use any combination of these conditions:
```pwsh
if (Get-Command scoop -ErrorAction Ignore) { ... }

$scooppath = if ($Env:SCOOP) { $Env:SCOOP } else { "~\scoop" }
if (Test-Path $scooppath -ErrorAction Ignore) { ... }
```

**Reply by bahyway** [reply]
Thank you for your response ..
but I wonder why then I got to get that "Scoop Exists" message?:

PS C:\Users\Me> scoop uninstall scoop
WARN  This will uninstall Scoop and all the programs that have been installed with Scoop!
Are you sure? (yN): y
Scoop has been uninstalled.
PS C:\Users\Me> if (Get-Command scoop -ErrorAction Ignore) { write-Host "Scoop already Exists." -ForegroundColor Yellow }
Scoop already Exists.
PS C:\Users\Me>

**Reply by bahyway** [reply]
Hi, even when I run the following I've got the same issue that I cannot really uninstall scoop when I check for its existence?:

PS C:\Users\Me> $scooppath = if ($Env:SCOOP) { $Env:SCOOP } else { "~\scoop" }
PS C:\Users\Me>  if (Test-Path $scooppath -ErrorAction Ignore) {scoop uninstall scoop } else { write-Host "Scoop does not Exist." -ForegroundColor Yellow }
WARN  This will uninstall Scoop and all the programs that have been installed with Scoop!
Are you sure? (yN): y
Scoop has been uninstalled.
PS C:\Users\Me> $scooppath = if ($Env:SCOOP) { $Env:SCOOP } else { "~\scoop" }
PS C:\Users\Me>  if (Test-Path $scooppath -ErrorAction Ignore) {scoop uninstall scoop } else { write-Host "Scoop does not Exist." -ForegroundColor Yellow }
WARN  This will uninstall Scoop and all the programs that have been installed with Scoop!
Are you sure? (yN): y
Scoop has been uninstalled.
PS C:\Users\Me>

**Reply by rashil2000** [reply]
What is the output of `Get-Command scoop`, and what is present at that location?
